pandas/core/arrays/datetimes.py

```diff
@@ -1785,6 +1852,32 @@ def _generate_range_overflow_safe(endpoint, periods, stride, side='start'):
     return other_end
 
 
+def _generate_range_recurse(endpoint, periods, stride, side):
+    """
+    Avoid problems in int64/uint64 mismatch by splitting range generation into
+    smaller pieces.
+
+    Parameters
+    ----------
+    endpoint : int
+    periods : int
+    stride : int
+    side : {'start', 'end'}
+
```

Comment:
so pls move all of this new code to the same place as checked_add_with_arr

Reply:
algos seems like a weird place to put datetime-specific code, no?  I guess in principle this code could be used for generating int64 ranges...